.github/workflows/main.yml

```diff
@@ -0,0 +1,138 @@
+name: Build and Deploy to a Single Device
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+env:
+  MENDER_SERVER_URL: https://hosted.mender.io           # Mender server URL
+  MENDER_DEPLOYMENT_NAME: my-app-v1.0-dev               # Mender deployment name
+  MENDER_RELEASE_NAME: change-the-world-to-better-v1.0  # Mender release name
+  MENDER_ARTIFACT_NAME: artifact.mender                 # Mender artifact name
+
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+    # conrainer with pre-installed mender-artifact and mender-cli tools
+    container:
+      image: mendersoftware/mender-ci-tools:mender-master
+    name: Upload a dummy Mender artifact
+    steps:
+      # Checkout git repository
+      - uses: actions/checkout@v3
+      # Download Mender bash client
+      - name: Setup Mender bash client
+        run: |
+          curl -O https://raw.githubusercontent.com/mirzak/bash-mender-client/d5a4ded71404f8526362f5b94e7f79065cfced09/mender-client.sh
+          chmod +x mender-client.sh
```

Comment:
You should use either the Docker image `mendersoftware/mender-client-bash:1.0.0` or at least our own copy of the bash client from [saas-tools repo](https://github.com/mendersoftware/saas-tools/blob/master/mender-client/mender-client.sh)

Reply:
Yes, I know that this is not the nicest way of doing it, but:
1) this is public repo and taking the bash client from the private `saas-tools` repo doesn't sound like the best thing to do
2) taking it from a docker container also requires not the best list of steps (docker pull, docker create, docker cp, docker rm)

I believe the best would be to open source the mender bash client and download it form there.

Reply:
Done, you can curl the client from here:
* https://github.com/mendersoftware/bash-mender-client

I'll follow-up independently in getting the latest fixes from our private repo to this one.